net/src/client/tcp/actors/jsonrpc.rs

```diff
@@ -14,73 +14,111 @@ use async_jsonrpc_client::{
 };
 use futures::StreamExt;
 use futures_util::compat::Compat01As03;
+use rand::seq::SliceRandom;
 use serde::Serialize;
-use serde_json::{value, Value};
+use serde_json::value;
+
+pub use serde_json::Value;
 
 use super::Error;
 
+const DEFAULT_BACKOFF_TIME_MILLIS: u64 = 250;
+
+struct Connection {
+    backoff: Duration,
+    socket: TcpSocket,
+    timestamp: Instant,
+    url: Arc<Mutex<String>>,
```

Comment:
Why does the `url` need to be an `Arc<Mutex<String>>` instead of a `String`? Is it to avoid trying multiple urls at the same time?

Reply:
`Arc<Mutex<T>>` is used extensively in the wallet component to safely share data across actors.

In this particular case, it helps other actors to read a reference to the URL currently being used depite the URL being changed internally by the JsonRpc actor at its own discretion.